File: UserProfileService/UserProfileService.Application/DTOs/ProfileDTOs/CreateProfileDTO.cs
using UserProfileService.Domain.Enums;

namespace UserProfileService.Application.DTOs
{
    public class CreateProfileDTO
    {
        public int? UserId { get; set; }
        public string Name { get; set; }
        public double Weight { get; set; }
        public double Height { get; set; }
        public DateOnly Birthday { get; set; }
        public Gender Gender { get; set; }
        public ActivityLevel ActivityLevel { get; set; }
    }
}

File: NutritionBalance.Client/src/app/models/profile/DTOs/profile/create-profile.dto.ts
import { ActivityLevel } from '../../enums/activity-level.enum';
import { Gender } from '../../enums/gender.enum';

export class CreateProfileDTO {
  userId?: number;
  name!: string;
  weight!: number;
  height!: number;
  birthday!: Date;
  gender!: Gender;
  activityLevel!: ActivityLevel;
}
